core/variant_op.cpp

```diff
@@ -457,24 +457,13 @@ void Variant::evaluate(const Operator &p_op, const Variant &p_a,
 			}
 
 			CASE_TYPE(math, OP_EQUAL, ARRAY) {
-				if (p_b.type != ARRAY) {
-					if (p_b.type == NIL)
-						_RETURN(false);
+				if (p_b.type != ARRAY)
 					_RETURN_FAIL;
-				}
+
 				const Array *arr_a = reinterpret_cast<const Array *>(p_a._data._mem);
 				const Array *arr_b = reinterpret_cast<const Array *>(p_b._data._mem);
 
-				int l = arr_a->size();
-				if (arr_b->size() != l)
-					_RETURN(false);
-				for (int i = 0; i < l; i++) {
-					if (!((*arr_a)[i] == (*arr_b)[i])) {
-						_RETURN(false);
-					}
-				}
-
-				_RETURN(true);
+				_RETURN(*arr_a == *arr_b);
 			}
```

Comment:
You say in your text that `==` in Godot 3 compares by reference, but the previous version here seems to compare by value each element in the array, whereas your new version here is comparing by reference. Is this breaking behaviour?

Reply:
Well, this is quite embarrassing, I would slap myself on the wrist. You are correct, the equality part of this PR needs to be adjusted.